### ISSUE TITLE
Echo Dot 3rd Gen - No Device Detection

### ISSUE THREAD

**Issue by AzonInc** [originallyus/node-red-contrib-alexa-local#72]
I bought a new Echo Dot 3 and it doesn't detect any of the devices which the 2nd generation detects.
The Nodes stay in detecting state.

Tested it on a fresh node-red installation too but without any success.

**Comment by PeteBear** [comment]
I am using node red to connect to alexa and find I need to re-build node red and restart HA then it works for a while but this appears to need to be done every 10 new commands or so - maybe a similar issue

**Comment by justmvg** [comment]
I'm experiencing the exact same problem with my 3rd gen Echo Dot. The node stays in discovery mode.

**Comment by ppontre** [comment]
I'm experiencing same behaviour here in Italy/Italian, with echo dot 3rd gen.
Alexa says to "push the button on the HUE bridge".....
Thanx for yr help & Regard
Paolo

**Comment by jnogues** [comment]
In spanish version with 3rd gen Echo Dot, don't discovers too.

**Comment by justmvg** [comment]
@hammondb4 mentioned the wemo emulator module in issue #47 and that module works great.
Just installed it using the palette by searching "node-red-contrib-wemo-emulator". After configuring the node, the echo dot discovers it right away after scanning.
Code can be found on https://github.com/gtalusan/node-red-contrib-wemo-emulator

**Comment by hammondb4** [comment]
Great you @justmvg could confirm it.
In my spanish locale echo spot and echo dot 3 wemo emulator is working ok.
Hope devs could use wemo working discovery method and implement it in alexa local....
I'm not an expert programer so would be nice if a project dev can confirm it could be done...

**Comment by ppontre** [comment]
Hi there, 
I also can confirm that wemo works, but it's NOT dimming.....
HUE std has ON, OFF, and dimming, while WEMO, afaik, has only on/off
Correct ?
Regards
Paolo

**Comment by justmvg** [comment]
I'm pretty sure you're right about that @ppontre. Hope some developer will fix this module for everyone using dimmed lights.

**Comment by wongnam** [comment]
@justmvg As i know that Echo cannot handle more than 16 Wemo emulator devices, that is the downward Wemo vs Hue Bridge.

**Comment by Z18Aplha** [comment]
> I bought a new Echo Dot 3 and it doesn't discover any of the devices which the 2nd generation discovers.
> The Nodes stay in discovery state.
> 
> Tested it on a fresh node-red installation too but without any success.

Same here!

**Comment by a-bailey** [comment]
What I did and I know it sucks... is to also connect my old echo dot 1st gen and everything works again the echo dot just has to be connected in the network for the device discovery to work

**Comment by gpriestland** [comment]
Same here - Echo dot 3.  During discovery the nodes show 'discovery' but don't get discovered

**Comment by kedzior-io** [comment]
I have 2nd generation and 3rd generation and yes I have to use 2nd generation to discover devices and when I try to trigger with 3rd generation one (in Spanish) it tells me that the devices doesn't respond.

**Comment by LEMD49** [comment]
FYI I have an ECHO DOT 2Gen bought in AMAZON USA and an ECHO SPOT bought in AMAZON SPAIN, both connected to a Spain Amazon account and the Alexa-local node works fine for me with only minor annoyance that the node.status stays yellow with message "discovery (p:xxx)".   Also the Echo Dot language is EN and Echo Spot is ES.

**Comment by wongnam** [comment]
Hi ALL,
I can use a Echo Dot 3rd Gen without a problem if i mix it with 2nd or 1st gen of echo dot.

**Comment by danielpondruff** [comment]
same behavior here. have 2x echo dot 3 in my network. 
alexa can't find devices. node stays in discovery mode

**Comment by kedzior-io** [comment]
Strange thing. I was about to return my 3rd gen for but it suddenly started to work. To all nodes in English I added equivalent in Spanish and used 2nd gen to "find devices". Seems to be working most of the time. Only sometime it goes "devices doesn't respond" (in Spanish) when I use 3rd gen. Same device works with 2nd gen.

**Comment by wongnam** [comment]
Re-check your Smarthome group in Alexa App. The << "devices doesn't respond" >> will be happen if sometime your device group in Alexa app is no device inside.

**Comment by toufou54** [comment]
Same for me, French version Dot gen 3, always in discovery mode yellow but Echo not found it

**Comment by Keltere** [comment]
@kedzior-io there is a pull request for the dot gen 3 fix. Please could you give a try?
I'm still trying to install node-red on my shitty sinology 214play so i can't test it yet.

https://github.com/originallyus/node-red-contrib-alexa-local/pull/85

**Comment by Keltere** [comment]
@danielpondruff don't know if you are talking with me. If you do just download [this file](https://raw.githubusercontent.com/originallyus/node-red-contrib-alexa-local/d9f6d3402f707898d7cf04fb921b923b356efdba/index.js) and place it on the directory where it's installed the node-red module  should be something like that `/usr/local/lib/node_modules/node-red-contrib-alexa-local/`
or just use `npm list -g` to check where it's installed the package.

**Comment by DevilDesire** [comment]
I experimented a little bit around.

First, I've added a new configuration parameter so you can set your desired port.

Then I looked at the wemo emulator, which gives more informations back to the echo 3rd gen.
Now i'm trying to do the same

**Comment by Keltere** [comment]
@DevilDesire hi, sorry to bother but did you find something?
Also thanks for suggesting wemo emulator, working just fine but missing feature like volume control.

**Comment by wongnam** [comment]
I read some where in Github that FYI.

> Echo Dot 3 gen is trying to connect to 80 even you send to it that the hub is running on a different port. Alexa-local is using ports greater that 1024 because of that is not working with newer Echo devices.

**Comment by wongnam** [comment]
Now i try to use node-red-contrib-amazon-echo for my Echo Dot 3rd gen while i am still waiting for originallyus to improve the Alexa Local node.

**Comment by wongnam** [comment]
@Keltere The 3rd option is to mix at least an Echo Dot 1st Gen or 2nd Gen into the same network(of cause same account).

**Comment by Barabba11** [comment]
for those who have difficulties with Echo Plus here is how I got it work:
https://github.com/datech/node-red-contrib-amazon-echo/issues/5#issuecomment-456573898

**Comment by lacrimosa2k2** [comment]
same here 3.gen dot doesnt work :(

**Comment by sapnho** [comment]
There are 388 weekly downloads, so there remains a high demand for this great node. I wrote a summary of my installation (https://www.thedigitalpictureframe.com/voice-control-digital-photo-frame-amazon-echo-mqtt-node-red/) which includes 2nd and 3rd generation Echo Dots and that works fine. But it would be great to get an update to make Gen 3 word out of the box. Thanks!

**Comment by wongnam** [comment]
Yep, the node is still working fine except it cannot handles over 20 devices.

**Comment by 877dev** [comment]
I just got a 3rd gen Echo and it's the same for me, just shows 'discovery' on the node.
Any updates?
Thx
